kymatio/scattering2d/backend/backend_torch.py

```diff
@@ -104,10 +104,12 @@ class Modulus(object):
         output: a tensor with imaginary part set to 0, real part set equal to
         the modulus of x.
     """
-    def __call__(self, input):
+    def __call__(self, x):
 
-        norm = input.norm(p=2, dim=-1, keepdim=True)
-        return torch.cat([norm, torch.zeros_like(norm)], -1)
+        norm = torch.zeros_like(x)
+        norm[...,0] = (x[...,0]*x[...,0] +
+                       x[...,1]*x[...,1]).sqrt()
```

Comment:
Could it be useful to do some parts inplace?

Either
```
norm[..., 0] = x[..., 0]
norm[..., 0] *= norm[..., 0]
norm[..., 0] += x[..., 1] * x[..., 1]
norm.sqrt_()
```

or

```
norm[:] = x
norm *= norm
norm[..., 0] += norm[..., 1]
norm.sqrt_()
```

Reply:
Yeah I thought about this. The risk is that splitting it up forces PyTorch to make each a separate kernel, which would slow things down. On the other hand, it may not. Worth benchmarking.

Reply:
Original:
```
modulus:            177.3 GB/s
```
Solution #1:
```
modulus:            99.84 GB/s
```
Solution #2:
```
modulus:            122.1 GB/s
```

So it looks like it's better to try to combine things, when possible.